lib/mix/test/mix/utils_test.exs

```diff
@@ -79,6 +79,16 @@ defmodule Mix.UtilsTest do
   end
 
   test "proxy_config reads from env and returns credentials" do
+    # Store what the environment is on
+    http_proxy = System.get_env("http_proxy")
+    hTTP_PROXY = System.get_env("HTTP_PROXY")
+    hTTPS_PROXY = System.get_env("HTTPS_PROXY")
```

Comment:
I'd call these variables `http_proxy`, `http_proxy_uppercase`, `https_proxy_uppercase` to avoid the hack of first lowercase letter and so on.

Reply:
Alternatively, do

```elixir
previous_env = %{
  "http_proxy" => System.get_env("http_proxy"),
  "HTTP_PROXY" => ...,
}

Enum.each(previous_env, fn {var, _value} -> System.delete_env(var) end)

# test...

after 
  System.put_env(previous_env)
```